src/Directions/Classes/DirectionsViaWaypoint.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+declare(strict_types=1);
+
+namespace GoogleMapsClient\Directions\Classes;
+
+use GoogleMapsClient\Classes\Geolocation;
+
+class DirectionsViaWaypoint
+{
+    /** @var Geolocation */
+    private $location;
+
+    /** @var int */
+    private $stepIndex;
+
+    /** @var float */
+    private $stepInterpolation;
+
+    public function __construct(Geolocation $location, int $stepIndex, float $stepInterpolation)
+    {
+        $this->location = $location;
+        $this->stepIndex = $stepIndex;
+        $this->stepInterpolation = $stepInterpolation;
+    }
+
+    public static function factory(\stdClass $stdViaWaypoint): self
+    {
+        return new self(
+            Geolocation::factory($stdViaWaypoint->location),
+            intval($stdViaWaypoint->step_index),
+            floatval($stdViaWaypoint->step_interpolation)
```

Comment:
Are you sure you want to store a float with `float`?

Reply:
you're right I will change it to string